bool enable
float64 meters
---